src/sqlfluff/core/rules/base.py

```diff
@@ -107,7 +107,20 @@ def __init__(self, edit_type, anchor, edit=None):
             raise ValueError("Unexpected edit_type: {0}".format(edit_type))
         self.edit_type = edit_type
         self.anchor = anchor
+        # Coerce to list
+        if isinstance(edit, BaseSegment):
+            edit = [edit]
         self.edit = edit
+        if self.edit:
+            # Copy all the elements of edit to stop contamination

```

Comment:
🚫 🦠

Reply:
I've added some more detailed comments here to explain why...

```
        # Copy all the elements of edit to stop contamination.
        # We're about to start stripping the position markers
        # of some of the elements and we don't want to end up
        # stripping the positions of the original elements of
        # the parsed structure.
        self.edit = copy.deepcopy(edit)
```